node/src/components/consensus/highway_core/evidence.rs

```diff
@@ -30,13 +44,29 @@ pub(crate) enum EvidenceError {
 pub(crate) enum Evidence<C: Context> {
     /// The validator produced two units with the same sequence number.
     Equivocation(SignedWireUnit<C>, SignedWireUnit<C>),
+    /// The validator endorsed two conflicting units.
+    Endorsements {
+        /// The endorsement for `unit1`.
+        endorsement1: SignedEndorsement<C>,
+        /// The unit with the lower (or equal) sequence number.
+        unit1: SignedWireUnit<C>,
+        /// The endorsement for `unit2`, by the same creator as endorsement1.
+        endorsement2: SignedEndorsement<C>,
+        /// The unit with the higher (or equal) sequence number, on a conflicting fork of the same
+        /// creator as `unit1`.
+        unit2: SignedWireUnit<C>,
```

Comment:
Could this unit have a lower sequence number? Is the assumption that `unit1` will always be the one higher in the chain?

Reply:
It would be more resilient if we checked which of the units has lower sequence number and use that as a "base".

Reply:
Would it?
The requirement for `Evidence::Endorsements` is that `unit1` is the one with the lower (or equal) sequence number, so that only `unit2` needs a corresponding `swimlane2`.

I don't think it would be more resilient to do the swap on the receiver side instead. Would you even add a `swimlane1`, and leave one of the two swimlanes always empty?

Reply:
There will always be only one swimlane, right? That "connects" the lower unit with the higher one.

We don't have to do this though but we should verify that `unit1` has lower sequence number than `unit2`.

Reply:
But we're implicitly doing that already: we check that the swimlane is a contiguous sequence of units by the same sender, leading from `unit2` back to something that "is an equivocation" together with `unit1`. The equivocation part checks that those have the same sequence number.